packages/identity-service/test/authenticationTest.js

```diff
@@ -321,4 +321,49 @@ describe('test authentication routes', function () {
     await updatedAuthRecord.destroy()
     await userRecord.destroy()
   })
+
+  it('skips otp for recognized devices', async function() {
+    const redis = app.get('redis')
+    const visitorId = 'abc123'
+    await signUpUser()
+    const userRecord = await models.User.findOne({ where: { email: 'dheeraj@audius.co' } })
+    await userRecord.update({ blockchainUserId: 1 })
+
+    await request(app)
+      .get('/authentication')
+      .query({
+        lookupKey:
+          '9bdc91e1bb7ef60177131690b18349625778c14656dc17814945b52a3f07ac77',
+        username: 'dheeraj@audius.co',
+        visitorId
+      })
+      .expect(403)
+
+    let fpRecord = await models.Fingerprints.findOne({ where: { visitorId } })
+    assert.strictEqual(fpRecord, null)
+
+    otp = await redis.get('otp:dheeraj@audius.co')
+
+    await request(app)
+      .get('/authentication')
+      .query({
+        lookupKey:
+          '9bdc91e1bb7ef60177131690b18349625778c14656dc17814945b52a3f07ac77',
+        username: 'dheeraj@audius.co',
+        visitorId,
+        otp,
+      })
+      .expect(200)
+
+    // validateFingerprint is called asynchronously and not awaited.
+    // This should be plenty of time since the dependency is mocked in test/lib/app.js
+    await new Promise(r => setTimeout(r, 200));
```

Comment:
would setImmediate here work then?

Reply:
Unfortunately setImmediate just puts a given function in a queue and the test doesn't wait up for it